### PR TITLE
ResourceListing provides a `first-item` convenience method

### PR DESCRIPTION
**Pull request by eyelidlessness**

### CODE REVIEW

**Comment by eyelidlessness** [comment]
Hold on this for a bit...

**Comment by eyelidlessness** [comment]
Added `:error-handler` support.

**Comment by eyelidlessness** [comment]
But it doesn't use the default `error-handler` yet. Fix incoming...

File: src/cljs_rest/core.cljs [hunk 1]
```diff
@@ -159,11 +159,38 @@
...
   (item-constructor [_])
   (items-constructor [_]))
 
 (defrecord ResourceListing [url opts item-opts constructor ok? data]
   RestfulListing
+  (first-item [this] (first-item this nil))
+
+  (first-item [this params]
+    (go
+      (let [resources (<! (read this params))
```
Comment (bwreilly) [inline src/cljs_rest/core.cljs]: `resources` is really the response right? Would it be more clear here and below named `response`?

Comment (eyelidlessness) [inline src/cljs_rest/core.cljs]: It's an instance of `ResourceListing`

**Pull request merged**